lib/internal/validators.js

```diff
@@ -48,6 +48,20 @@ function validateAndMaskMode(value, name, def) {
   throw new ERR_INVALID_ARG_VALUE(name, value, modeDesc);
 }
 
+function validateInteger(value, name) {
+  let err;
+
+  if (typeof value !== 'number')
+    err = new ERR_INVALID_ARG_TYPE(name, 'number', value);
+  else if (!Number.isInteger(value))
```

Comment:
Shall we maybe test for `Number.isSafeInteger`?

Reply:
I have mixed feelings. It seems like `isSafeInteger()` would be... safer... but I also don't want to artificially limit people if they know what they are doing (see original issue). It's probably a pretty corner case to need a value that large too.

Reply:
If someone passes a number that is that big it is clearly an error because no file system would be able to handle such big values.

Reply:
I'm not sure what the maximum file size on every file system out there is, but I don't see much of a point in throwing on `2**53`, but allowing `2**53 - 1`. Hypothetically, these validators could be reused for other things not related to `fs` as well.

Reply:
Even if we reuse the validator, it should normally validate for a safe integer. Anything above will always cause issues.

Reply:
@cjihrig we actually already use `Number.isSafeInteger` in the fs streams. Using it here as well would be coherent.